pom.xml

```diff
@@ -288,6 +288,37 @@
 				</plugins>
 			</build>
 		</profile>
+                <profile>
+            		<id>java9-plus</id>
+            		<activation>
+             		   	<jdk>[9,)</jdk>
+            		</activation>
+            		<dependencies>
+                		<dependency>
+                    			<groupId>javax.xml.ws</groupId>
+                    			<artifactId>jaxws-api</artifactId>
+                    			<version>2.3.0</version>
+                		</dependency>
+                                <dependency>
+                    			<groupId>com.sun.xml.bind</groupId>
+                    			<artifactId>jaxb-impl</artifactId>
```

Comment:
We should be using `org.glassfish.jaxb:jaxb-runtime` instead of `com.sun.xml.bind:jaxb-*`.

Reply:
Hi @zregvart,

Thanks for the input, but I hesitate to do it for now. Because org.glassfish.jaxb:jaxb-runtime isn't OSGi bundle yet while com.sun.xml.bind:jaxb-*. is. We still need OSGi bundle for Karaf runtime, and we want all jaxb impl align across all components used in FUSE.

Why you eager to use org.glassfish.jaxb for now? Is it because it resolved the split package issue(which IMHO isn't a big deal since we use classpath compatible way at this stage.)? Or some other issues force you to use org.glassfish.jaxb:jaxb-runtime instead of com.sun.xml.bind:jaxb-*?

Thanks!
Freeman

Reply:
As far as I understand the JAXB implementation will ship with [those Maven coordiantes](https://javaee.github.io/jaxb-v2/doc/user-guide/ch03.html#deployment-maven-coordinates). I'm ever weary of perpetuating the use of `com.sun.xml` artefacts as I'm not sure when will new releases stop under those coordinates.

I guess now that JAXB is at Eclipse, those coordinates will again change to `jakarta.*` so we'll probably need to switch to those.

So not sure what is the correct solution here, staying on `com.sun.xml`, going with `org.glassfish.jaxb` or waiting for `jakarta.*` artefacts to surface. For me staying with `com.sun.xml` feels wrong, going with `org.glassfish.jaxb` feels slightly less wrong, going with `jakarta.*` artefacts would be the correct solution -- but those artefacts are not available on Maven Central (I've found a [staging repository](https://oss.sonatype.org/content/groups/staging/jakarta/xml/bind/jakarta.xml.bind-api/2.3.2/) with just the API artefact, and it has OSGI metadata).

Also, of interest but not a hard requirement, Spring (Boot) [is using](https://github.com/spring-projects/spring-boot/blob/60b35df2159e6e44a2031e65979e30f6ce7f301d/spring-boot-project/spring-boot-dependencies/pom.xml#L1989-L1993) `org.glassfish.jaxb`.

Reply:
Let's go with org.glassfish.jaxb, I really wanna get rid of the split package issue.
I will wrap org.glassfish.jaxb in servicemix-bundles project, to make it OSGi ready.